File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.cs
using System;

public class Solution {
    public int[,] solution(int n) {
        int[,] answer = new int[n,n];
        int count = 0;
        bool reverse = false;
        int startX =0;
        int startY =0;
        int maxX = n;
        int maxy = n;
        int minX =0;
        int minY =0;
        while(count < n * n)
        {
            if(reverse == false)
            {
                for(int i = startX ; i < maxX; i++)
                {
                    answer[startY,i] = ++count;
                    startX = i;
                }
                 for(int j = startY + 1 ; j < maxy; j++)
                {
                    answer[j,startX] = ++count;
                    startY = j;
                }
                maxX--;
                maxy--;
            }
            else
            {
                for(int i = startX - 1 ; i >= minX; i--)
                {
                    answer[startY,i] = ++count;
                    startX = i;
                }
                for(int j = startY - 1 ; j > minY; j--)
                {
                    answer[j,startX] = ++count;
                    startY = j;
                }
                minX++;
                minY++;
                startX++;
            }
            reverse = !reverse;
        }
        return answer;
    }
}

File: 프로그래머스/0/181832. 정수를 나선형으로 배치하기/정수를 나선형으로 배치하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(int n) {
    vector<vector<int>> answer(n,vector<int>(n));
    int count = 1;
    int top = 0;
    int bottom = n - 1;
    int left = 0;
    int right = n - 1;
    
    while (top <= bottom && left <= right) {
        for (int j = left; j <= right; ++j) {
            answer[top][j] = count++;
        }
        top++;

        if (top > bottom) break;

        for (int i = top; i <= bottom; ++i) {
            answer[i][right] = count++;
        }
        right--;

        if (left > right) break;

        for (int j = right; j >= left; --j) {
            answer[bottom][j] = count++;
        }
        bottom--;

        if (top > bottom) break;

        for (int i = bottom; i >= top; --i) {
            answer[i][left] = count++;
        }
        left++;
    }

    return answer;
}